rio_tiler/models.py

```diff
@@ -391,6 +406,15 @@ def apply_color_formula(self, color_formula: Optional[str]):
     def apply_expression(self, expression: str) -> "ImageData":
         """Apply expression to the image data."""
         blocks = get_expression_blocks(expression)
+
+        stats = self.dataset_statistics
+        if numpy.array(stats).all() is not None:
+            res = []
+            for prod in itertools.product(*stats):  # type: ignore
+                res.append(apply_expression(blocks, self.band_names, numpy.array(prod)))
+
+            stats = list(zip([min(r) for r in zip(*res)], [max(r) for r in zip(*res)]))
```

Comment:
This is the only real interesting part of the addition (because technically you can access the metadata directly with rasterio). Here we calculate the result of the expression on the min/max values so we get the min/max of the expression output